s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/input/JsonTransformer.java

```diff
@@ -47,30 +50,13 @@ public void configureValueConverter(final Map<String, String> config, final S3So
     }
 
     @Override
-    public List<Object> getRecords(final InputStream inputStream, final String topic, final int topicPartition,
-            final S3SourceConfig s3SourceConfig) {
-        final List<Object> jsonNodeList = new ArrayList<>();
-        JsonNode jsonNode;
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
-            String line = reader.readLine();
-            while (line != null) {
-                line = line.trim();
-                if (!line.isEmpty()) {
-                    try {
-                        // Parse each line as a separate JSON object
-                        jsonNode = objectMapper.readTree(line.trim()); // Parse the current line into a JsonNode
-                        jsonNodeList.add(jsonNode); // Add parsed JSON object to the list
-                    } catch (IOException e) {
-                        LOGGER.error("Error parsing JSON record from S3 input stream: {}", e.getMessage(), e);
-                    }
-                }
-
-                line = reader.readLine();
-            }
+    public Stream<Object> getRecords(final IOSupplier<InputStream> inputStreamIOSupplier, final String topic,
+            final int topicPartition, final S3SourceConfig s3SourceConfig) {
+        try {
+            return readJsonRecordsAsStream(inputStreamIOSupplier);
         } catch (IOException e) {
-            LOGGER.error("Error reading S3 object stream: {}", e.getMessage());
+            throw new RuntimeException("Error in parsing json records", e); // NOPMD rethrowing an exception
```

Comment:
I think this should probably be a ConnectException (which extends RunTimeException.).

Reply:
Keeping it similar to other transformers, handling the exception while reading the stream